File: longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int n=s.size(),x=0,y=0;
        vector<vector<bool>>v(n,vector<bool>(n,false));
        for(int gap=0;gap<n;gap++){
            int i=0,j=gap;
            while(j<n){
                if(gap==0){
                    v[i][j]=true;
                }
                else if(gap==1){
                    if(s[i]==s[j]){
                        v[i][j]=true;
                    }
                }
                else{
                    if(s[i]==s[j]&&v[i+1][j-1])
                        v[i][j]=true;
                }
                if(v[i][j]){
                    x=i;
                    y=j;
                }
                
                i++;
                j++;
            }
        }
        string ans="";
        if(x>y){
            swap(x,y);
        }
        for(int i=x;i<=y;i++){
           ans+=s[i];
        }       
        return ans;
    }
};

File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        ans = s[0]
        n = len(s)
        for k in range(n):
            i = k-1
            j = k+1
            while i>=0 and j<n and s[i]==s[j]:
                if len(ans)<len(s[i:j+1]):
                    ans = s[i:j+1]
                i-=1
                j+=1
            i = k
            j = k+1
            while i>=0 and j<n and s[i]==s[j]:
                if len(ans)<len(s[i:j+1]):
                    ans = s[i:j+1]
                i-=1
                j+=1
        return ans